### ISSUE TITLE
secret in commands outputed to logs

### ISSUE THREAD

**Issue by amolari** [F5Networks/f5-bigip-runtime-init#35]
I need to set the device master key. Constrains: the master-key value must not be shown in the logs (it's a secret)

My config is:
```
bigip_ready_enabled:
  - name: set_message_size
    type: inline
    commands:
      - '/usr/bin/curl -s -f -u admin: -H "Content-Type: application/json" -d ''{"maxMessageBodySize":134217728}'' -X POST http://localhost:8100/mgmt/shared/server/messaging/settings/8100 | jq .'
      - f5mku -r $(echo -n "{{{ MASTER_KEY }}}" | base64)
```
I see in the logs:
```
2021-12-07T12:24:23.774Z [17952]: info: Executing custom bigip_ready_enabled commands
2021-12-07T12:24:23.774Z [17952]: info: Executing inline shell command
2021-12-07T12:24:23.775Z [17952]: debug: command: /usr/bin/curl -s -f -u admin: -H "Content-Type: application/json" -d '{"maxMessageBodySize":134217728}' -X POST http://localhost:8100/mgmt/shared/server/messaging/settings/8100 | jq .
2021-12-07T12:24:23.816Z [17952]: info: Shell command: One or more commands contains a secret, skipping logging command and output
2021-12-07T12:24:23.817Z [17952]: debug: /usr/bin/curl -s -f -u admin: -H "Content-Type: application/json" -d '{"maxMessageBodySize":134217728}' -X POST http://localhost:8100/mgmt/shared/server/messaging/settings/8100 | jq . execution completed; response: {
  "maxMessageBodySize": 134217728,
  "localhostRestnodedConnectionLimit": 8,
  "defaultEventHandlerTimeoutInSeconds": 60,
  "minEventHandlerTimeoutInSeconds": 15,
  "maxEventHandlerTimeoutInSeconds": 60,
  "maxActiveLoginTokensPerUser": 100,
  "generation": 1,
  "lastUpdateMicros": 1638879863812231,
  "kind": "shared:server:messaging:settings:8100:restservermessagingpoststate",
  "selfLink": "https://localhost/mgmt/shared/server/messaging/settings/8100"
}

2021-12-07T12:24:23.817Z [17952]: info: Executing inline shell command
2021-12-07T12:24:23.817Z [17952]: debug: command: f5mku -r $(echo -n "abcdef" | base64)
2021-12-07T12:24:23.872Z [17952]: error: Command failed: f5mku -r $(echo -n "abcdef" | base64)
Error trying to rekey: 01071029:5: Invalid master key
```

The secret is recognized ("_One or more commands contains a secret, skipping logging command and output_") as such but its value is outputed in the logs.
Could that be improved (whatever the log level)?

Remarks:
1. `f5mku -r` is also used in some examples in this repo. It seems that it doesn't work as expected (as observed in previous output;I've tested with v14.1.4). F5 Support doesn't recommand the use of `f5mku` to set the master key but the tmsh command `tmsh modify sys crypto master-key prompt-for-password` (ref: [K73034260: Overview of the BIG-IP system Secure Vault feature](https://support.f5.com/csp/article/K73034260))
2. to set the master key with tmsh I would use the command `expect -c 'log_user 0; spawn tmsh modify sys crypto master-key prompt-for-password; expect "assword:"; send "{{{ MASTER_KEY }}}\r"; expect "assword again:"; send "{{{ MASTER_KEY }}}\r"; interact'` but have of course the same issue of secret outputted to logs.

**Comment by f5-applebaum** [comment]
Logs with secrets are not suppressed in debug (logLevel: silly) as sometimes needed in debugging/troubleshooting interpolation issues, troubleshooting adjacent commands, etc.  However, agree. Should have the ability to suppress on all levels no matter what and enabled by default. 

example:
```
controls:
  logLevel: silly
  logFilename: /var/log/cloud/bigIpRuntimeInit.log
  logSuppressSecretsInDebug: true
extension_packages: 
...
```

_Disclaimer: final attribute name tbd. Just example of log control._

Re: Remarks. 
In general, yes, it's always best practice to go through tmsh/gui when possible ( as it is the main supported api with mcp validation, error handling, internal commands can be removed at any time, etc ). However, it is documented and this is a pretty limited use case ( initial onboarding ). Will discuss with Support if extra complexity going through expect/tmsh is required in this particular use case (or more that it potentially may be removed) and have examples updated accordingly. Thanks!

**Comment by alexapplebaum** [comment]
Secrets will be suppressed at all levels. Addressed in ESECLDTPLT-3028.

**Comment by shyawnkarim** [comment]
Closing.

This ticket was included with [Release 1.4.2](https://github.com/F5Networks/f5-bigip-runtime-init/releases/tag/1.4.2).

**Comment by amolari** [comment]
@shyawnkarim Please add this setting in the documentation (https://github.com/F5Networks/f5-bigip-runtime-init#controls)
